dev/package-examples/nginx-1.2.0/manifest.yml

```diff
@@ -12,3 +12,45 @@ requirement:
     versions: ">7.1.0 <7.6.0"
   elasticsearch:
     versions: ">7.0.1"
+
+datasources:
+  -
+    # Do we need a name for the data source?
+    name: nginx
+
+    # List of inputs this datasource supports
+    inputs:
+      -
+        # An id can be given, in case the type used here is not unique
+        # This is for selection in the stream
+        # id: nginx
+        type: metrics/nginx
+
+        # Common configuration options for this input
+        vars:
+          - name: hosts
+            description: Nginx hosts
+            default:
+              ["http://127.0.0.1"]
+            # All the config options that are required should be shown in the UI
+            required: true
+          - name: period
+            description: "Collection period. Valid values: 10s, 5m, 2h"
+            default: "10s"
+          - name: username
+            type: text
+          - name: password
+            # This is the html input type?
+            type: password
```

Comment:
I think we need to decide all possible values for `type` field. They do not need to be HTML types, though. Here are the possible types I currently see (or can foresee being useful): 
```
text -> maps to UI text input
number -> maps to UI number input
boolean -> maps to UI checkbox
duration -> maps to UI text input
  this would let us do client-side validation but maybe overkill for now?
url -> maps to UI text input
  this would let us do client-side validation but maybe overkill for now?
```
In the case of password vars, I would use `text` for them as they'll show up as simple text inputs in the UI.

Reply:
I like the above proposed types. I was thinking of password instead of text to not have to show clear text passwords directly on the screen when users put them in.

Reply:
BTW: If we have these predefined values for type we can also validate it already on the package side to make sure nothing odd shows up in the package itself.

Reply:
For the _URL_ types we have to be careful when we do this and we do it, in some modules we do use compound scheme (not sure if this is the exact term?) Like "http+npipe://./pipe/custom" see https://github.com/elastic/beats/blob/d75261469b5c2e675d7d331b86bfc8599d797ddc/metricbeat/mb/parse/url_test.go for a few examples.

Reply:
I see url support as a stretch goal. In case of doubt we can always fall back to just string. Lets keep it simple for now.

Reply:
We can add granular text types (duration, url, etc) but it doesn't mean that we need to do validation right now. Validation can be added later on from all fronts (UI, agent, package) once we align on rules.

Reply:
Also ++ on keeping password type. This give the UI information to decide when to show clear text inputs and when to obfuscate.

Reply:
As soon as we started the UI implementation and are happy with the format, we should open a PR here with the exact specs.